### DISCUSSION TITLE
add func to check if project is exported

### DISCUSSION THREAD

**Discussion by azur-wolve** [godotengine/godot-proposals#4711] [Scripting]
`is_project_exported() -> bool `
Having it to check at run-time if the project is an exported one, or not (in-editor debug), could be useful for conditional code execution.
(code that works in editor but not in exported and the reversal, and also for stuff that only should be accessible on "dev mode")

Recently stumbled with a problem:
after exporting a project, it doesn't has the directories, so when checking if a file exists at a give path (which works well on editor) will suppose a run-time crash on an exported one.

In the editor this works well:
```
    var dir_path := "res://parent_dir/target_dir/"
    var dir:= Directory.new()

    if not dir.dir_exists(dir_path):
        dir.make_dir_recursive(dir_path)
```
But it wont work when the project its exported because. 
(the directories and all the stuff are packed in the `.pck` file, and don't know how to access them to know if some file or dir exists, how load something from there, etc)

Otherwise, if change the `make_dir_recursive()` line to:
```
    var dir_path := "parent_dir/target_dir/"      # here erase "res://" or will create 3 dirs → "res:" → "parent_dir" → "target_dir"
    var dir:= Directory.new()

    if not dir.dir_exists(dir_path):
        dir.make_dir_recursive( OS.get_executable_path().get_base_dir().plus_file(dir_path) )  # here is the matter
```
Now it works when the project is exported, but not on the editor.

Unless I'm missing or doing something wrong, the ability to branch it using any hypothetical `is_project_exported()` would solve it easily.

**Comment by Lielay9** [comment]
Isn't this what feature tags are for? [https://docs.godotengine.org/en/stable/tutorials/export/feature_tags.html](https://docs.godotengine.org/en/stable/tutorials/export/feature_tags.html)
There's also other useful stuff you can check with them, such as the platform, or even create your own.
